### PR TITLE
glib-networking: gnutls --without-p11-kit req

### LINKED ISSUE #36323: Glib-Networking fails to build on 10.9
Gists: https://gist.github.com/560430c35c83553371b3

Important section:

```
libtool: link: /usr/bin/grep -E -e "^g_io_module_(load|unload|query)" ".libs/libgiognutls.exp" > ".libs/libgiognutls.expT"
libtool: link: mv -f ".libs/libgiognutls.expT" ".libs/libgiognutls.exp"
libtool: link: sed -e 's,^,_,' < .libs/libgiognutls.exp > .libs/libgiognutls-symbols.expsym
libtool: link: clang  -o .libs/libgiognutls.so -bundle  .libs/gnutls-module.o .libs/gtlsbackend-gnutls.o .libs/gtlscertificate-gnutls.o .libs/gtlsclientconnection-gnutls.o .libs/gtlsconnection-gnutls.o .libs/gtlsdatabase-gnutls.o .libs/gtlsfiledatabase-gnutls.o .libs/gtlsinputstream-gnutls.o .libs/gtlsoutputstream-gnutls.o .libs/gtlsserverconnection-gnutls.o .libs/gtlsbackend-gnutls-pkcs11.o .libs/gtlscertificate-gnutls-pkcs11.o .libs/gtlsdatabase-gnutls-pkcs11.o   -Wl,-force_load,../../tls/pkcs11/.libs/libgiopkcs11.a  -L/usr/local/Cellar/glib/2.42.1/lib -L/usr/local/opt/gettext/lib -L/usr/local/Cellar/p11-kit/0.18.4/lib -lp11-kit -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lintl -L/usr/local/Cellar/gnutls/3.3.11/lib -lgnutls  -O2   -Wl,-exported_symbols_list,.libs/libgiognutls-symbols.expsym
Undefined symbols for architecture x86_64:
  "_gnutls_pkcs11_privkey_import_url", referenced from:
      _g_tls_certificate_gnutls_pkcs11_copy in gtlscertificate-gnutls-pkcs11.o
  "_gnutls_pkcs11_privkey_init", referenced from:
      _g_tls_certificate_gnutls_pkcs11_copy in gtlscertificate-gnutls-pkcs11.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [libgiognutls.la] Error 1
make[2]: *** [install] Error 2
make[1]: *** [install-recursive] Error 1
make: *** [install] Error 2
```

### PR DESCRIPTION
**Pull request by dunn**
Fixes #36323.

### CODE REVIEW

File: Library/Formula/glib-networking.rb [hunk 1]
```diff
@@ -11,6 +23,8 @@ def pour_bottle?
     false
   end
 
+  # see https://github.com/Homebrew/homebrew/issues/36323
+  depends_on GnutlsWithoutP11Requirement
```
Comment (mikemcquaid) [inline Library/Formula/glib-networking.rb]: Does `depends_on "gnutls" => "without-p11-kit"` work?

Comment (DomT4) [inline Library/Formula/glib-networking.rb]: AFAIK that syntax doesn't work. I thought we'd tried it before in places. Could be wrong though.

Comment (mikemcquaid) [inline Library/Formula/glib-networking.rb]: If it doesn't work: it should :grinning:

Comment (dunn) [inline Library/Formula/glib-networking.rb]: Yeah, it doesn't currently :(

**Comment by tdsmith** [comment]
Alex and I talked about this on slack; the "real" fix is to convince gnutls to actually build against p11-kit if it's available. glib-networking assumes that gnutls is built against p11-kit if both are available (p11-kit is visible to glib-networking because Homebrew regards it as a dependency of gnutls --with-p11-kit) but gnutls is presently refusing to build against p11-kit because the version we carry is too old.

**Comment by dunn** [comment]
And I ran into this while attempting to update p11-kit: https://bugs.freedesktop.org/show_bug.cgi?id=91602

**Comment by dunn** [comment]
Found a simpler fix; but I'll look into making `depends_on "formula" => "without-option"` work too.

**Comment by dunn** [comment]
Oh, right, the sandbox issue.

**Comment by DomT4** [comment]
And to finish tying up all the `glib-networking` related sandbox issues in Github history, ref #43165 :smile_cat:.

**Pull request closed without merging**